src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -51,21 +45,24 @@ export function DiscoverMainRoute(props: Props) {
   const {
     core,
     chrome,
-    uiSettings: config,
     data,
     toastNotifications,
     http: { basePath },
     dataViewEditor,
   } = services;
+  const { id } = useParams<DiscoverLandingParams>();
+  const stateContainer = useSingleton<DiscoverStateContainer>(() =>
+    getDiscoverStateContainer({
+      history,
+      services,
+    })
+  );
```

Comment:
Apart from adding a new state container, this is the biggest change, before this, `DiscoverStateContainer` was initialized 2x on every change to different saved search